### DISCUSSION TITLE
February release - We've launched our new API, updated our table component, added bindings to automations, and more

### DISCUSSION THREAD

**Discussion by joebudi** [Budibase/budibase#4949] [Announcements]
Spring - it’s often a time of newness, growth, and change… and that couldn’t be more true for us here at Budibase. As Budibase continues to grow in users and usage - since November, over 17,000 companies are now using Budibase - we decided to dedicate February to improving the extensibility and usability of Budibase.  

## Budibase API
You have asked - we have listened. The all-new documented Budibase API allows you access to a full HTTP interface where you can perform CRUD operations on applications, rows, users, and much more. This also comes with an OpenAPI3 specification that you can import directly into postman and other platforms.

To summarize, the new Budibase API enables:
- Budibase as a backend
- Extending apps
- Inter-operability

The Budibase rest API is fully token-based, and users are assigned authentication tokens that they use in calls to the API. This is fully RBAC aware and will prohibit users with the wrong roles from accessing data.

**Documentation**
Interactive documentation - [Visit interactive docs](https://docs.budibase.com/reference/post_applications)
General Budibase API documentation - [Visit docs](https://docs.budibase.com/docs/public-api)
Example: Build a CRUD app with Budibase and Next JS - [Visit Example](https://budibase.com/blog/building-a-crud-app-with-budibase-and-next.js/)

![Start building with Budibase's API (3)](https://user-images.githubusercontent.com/49767913/158996460-0ce4fed6-a129-482d-a39a-0cdf90540c44.png)

## Advanced Automation Section Configuration with Bindings
The Budibase automation section now can be controlled completely with handlebars bindings. This provides a level of flexibility not seen before in the Budibase automations section - allowing users to work with relationships as well as any other data types that Budibase supports.
<img width="552" alt="154073380-0683a2e0-918e-4d93-8ed9-72576c463cba" src="https://user-images.githubusercontent.com/49767913/158996357-69e65f37-9714-4d33-909c-3e4daf565490.png">

## Table component v2
The table component is one of our most heavily used components, and it was long overdue an update. So, we decided to extend its functionality by adding the following features:

- Format cell values using a handlebars expression
- Edit column styling properties - width, text-align, height, background color, and text color.
- Rearrange columns in your tables with an intuitive drag and drop interface.
- Allow/disallow selection of multiple rows (table checkboxes)
- Use the selected row in the table in other parts of your app.

Learn more about the [new table component](https://docs.budibase.com/docs/table)

![154986347-10f2bff9-1449-4f60-8889-b54bb711392d](https://user-images.githubusercontent.com/49767913/158996377-ac52536c-3594-4425-9f8e-cf21a306c2d6.png)

## Export JSON/CSV from client applications
Budibase users can now export JSON and CSV files from tables in their applications, and not just inside the builder. You can have end users export and download your orders/sales or any other data you store in your Budibase tables!

## New Budibase Documentation
The most common piece of feedback we’ve received from our community is we must improve our documentation. I am happy to announce we’ve updated our documentation site. The design is cleaner, and the content has been updated. This is just the beginning, and we will invest more time and effort into the documentation as we move forward. 
[Check out the new documentation](https://docs.budibase.com/docs)

## Additional, important improvements

- [Various SQL fixes](https://github.com/Budibase/budibase/pull/4795)
- [Allowing iframes from HTTPS URLs](https://github.com/Budibase/budibase/pull/4769)

## Check out the release video:
https://youtu.be/T-SOLirXi60

## Coming up in the March Release
- [Google sheets integration (we're still waiting on Google approval)](https://github.com/Budibase/budibase/pull/4089)
- [Map component](https://github.com/Budibase/budibase/pull/4856)
- Updated template section

## Community request
After analyzing feedback, we know there are two areas we need to improve on:
1. Docs
2. The UX around designing apps

We've made good progress on documentation, and now it's time to improve the design experience within Budibase. We are currently creating mockups for the new design section. If you'd like to see the new designs and provide feedback, please comment below and we'll organize a time for you to talk with our designer.

Have a great day Budis!

**Comment by Fabian1410** [comment]
Missing File Attachments in the api :)

**Reply by KernOil** [reply]
agreed much needed. attachments for MMSQL

**Comment by theobouwman** [comment]
How do I update my busibase system when using docker compose?
